apps/frontend/src/pages/CreateListing.tsx

```diff
@@ -1,5 +1,281 @@
+import Carousel from "../extra_components/Carousel.tsx";
+
+interface URLObject {
+  url: string;
+}
+interface LocationObject {
+  latitude: number;
+  longitude: number;
+}
+
+interface ListingObject {
+  title: string;
+  description: string;
+  price: number;
+  location: LocationObject;
+  images: URLObject[];
+}
+
+interface NewListingObject {
+  listing: ListingObject;
+}
+
 const CreateListing = () => {
-  return <h1>CreateListing</h1>;
+  const [images, setImages] = useState<string[]>([]);
+
+  // This newListingObject bundles all the listing data for upload to the server
+  const [newListingObject, setNewListingObject] = useState<NewListingObject>({
+    listing: {
+      title: "",
+      description: "",
+      price: 0,
+      location: {
+        latitude: 0.0,
+        longitude: 0.0,
+      },
+      images: [],
+    },
+  });
+
+  // Gets the user location, and adds it to the listing object
+  const getUserLocation = useCallback(async () => {
+    try {
+      navigator.geolocation.getCurrentPosition((position) => {
+        newListingObject.listing.location.longitude = position.coords.longitude;
+        newListingObject.listing.location.latitude = position.coords.latitude;
+      });
+      return true;
+    } catch (error) {
+      return false;
+    }
+  }, [newListingObject.listing.location]);
+
+  // Updates and sends the newListingObject, to the server via post under /api/listing
+  const sendPostToCreateListing = async () => {
+    // In order to make sure that the images are retrieved before submitting
+    const successImages = await uploadImages();
+    let successLocation: boolean = true;
+    if (newListingObject.listing.location.latitude !== 0) {
+      successLocation = false;
+      getUserLocation().then(() => {
+        successLocation = true;
+      });
+    }
+    //console.log("Image Successful", imageUploadSuccessful)
+    console.log("New Listing Object", newListingObject);
+    if (successImages && successLocation) {
+      _axios_instance
+        .post("/listing", newListingObject)
+        .then((response) => {
+          alert("Listing Created!");
+          console.log("Response to upload", response);
+        })
+        .catch((error) => {
+          console.log("Response to error", error);
+          alert("Listing Creation Failed");
+        });
+    } else if (!successImages) {
+      alert("Images failed to upload, please try again later");
+    } else {
+      // Currently this is added to catch if the location is not set, we could default this to the location of the university instead
+      alert(
+        "Error occurred when creating a listing, you may need to enable location permissions for this site",
+      );
+    }
+  };
+
+  // This function makes sure that the passed in url is a base64 data string
+  const isImageValid = (url: string) => {
+    try {
+      return url.startsWith("data");
+    } catch (error) {
+      return error === TypeError && url === undefined;
```

Comment:
If this is meant to return true when a url is undefined, that logic should be in the try block and not in the catch section along with an error. Returning true on an undefined url is also not ideal though because it's not a valid image, ideally intentionally undefined urls should be handled elsewhere in the code and this should return false on anything but a valid image url.